### PR TITLE
[MAVLINK] Add new mavlink messages

### PR DESCRIPTION
**Pull request by machadofelipe**
Hi devs, this adds a new mavlink data stream at 1Hz. This is crucial to have system messages for systems such as OpenHD.

1. Update throttle to also show navigation thr;
2. Send battery cell count and consumption in battery status;
3. Send baro or imu temperature in scaled pressure sensor;
4. Send same OSD system msg as status text.

Please, let me know if I can improve this commit. The messages are based on OSD functions and a bit duplicated, but I saw no better way to avoid messing with the OSD.

If possible, let me know your thoughts about updating iNav to the new Mavlink v2 and also including some msgs from ardupilot dialect. That way we could send ESC temperatures and Wind estimation via Mavlink.

### CODE REVIEW

File: src/main/telemetry/mavlink.c [hunk 1]
```diff
@@ -200,6 +223,311 @@ APM_PLANE_MODE inavToArduPlaneMap(flightModeForTelemetry_e flightMode)
     }
 }
 
+// Used twice, make sure it's exactly the same string
+// to save some memory
+#define RC_RX_LINK_LOST_MSG "!RC RX LINK LOST!"
+
+static const char * mavlinkArmingDisabledReasonMessage(void)
```
Comment (digitalentity) [inline src/main/telemetry/mavlink.c]: I think this PR would be the right time to unify the status text generation - it's shared between OSD and MAVLINK anyway. WDYT?

**Review by digitalentity — CHANGES_REQUESTED** [review]

File: src/main/telemetry/mavlink.c (hunk 1 above)
Comment (machadofelipe) [reply src/main/telemetry/mavlink.c]: Yes, sure, makes sense. Will do the changes them over the next weekend. This week I am out flying iNav in summer camping 😉

**Comment by machadofelipe** [comment]
Updated the PR. Please review and let me know if I can improve.
Tested both in bench as in the air with MATEKF405SE + QOpenHD and wifibroadcast-osd

Below is the full list of changes:
- Did the integration with the OSD system messages. For now the added Mavlink Status Text is limited to boards with OSD active, but in a short future I will try to generalize this part a bit more.
- Break down the failsafe mode to map real ardu modes. When no Receiver is active and mode in iNav is only "Failsafe", there is no real correpondent mode in Ardupilot.
- In the future it maybe better to do a getMode just for mavlink and find the correct mode without using mapping.
- Added sensors health report. Very useful for alarms in ezwifibroadcast and old osd in OpenHD.
- Correct heading to centidegrees added velocity per axis in global_position, as required from mavlink lib.
- Throttle telemetry reflects the signal used to command (includes navigation throttle).
- Added message to send battery status with cell average, number of cells, current consumed and energy consumed.
- Added baro temperature if available or send Imu temp. instead.
- For the new messages, a new telemetry stream (extra3) was added.
- Mavlink library update to latest release in v1.0 (better documented and less magic numbers overall)

**Comment by machadofelipe** [comment]
@digitalentity should I change anything to help the review process?

**Comment by digitalentity** [comment]
Sorry, this fell through the cracks. Will have a look later on today.

**Review by digitalentity — APPROVED** [review]
LGTM from me. @fiam can you have a second look at the changes in the OSD messages code?

**Pull request merged**